### PR TITLE
Export the TestActions on the TestObject

### PR DESCRIPTION
**Pull request by wolfs**
to make them accessible via the REST api.

### CODE REVIEW

File: src/main/java/hudson/tasks/test/TestObject.java [hunk 1]
```diff
@@ -217,6 +218,7 @@ public AbstractTestResultAction getTestResultAction() {
      * Get a list of all TestActions associated with this TestObject. 
      */
     @Override
+    @Exported(visibility = 999)
```
Comment (daniel-beck) [inline src/main/java/hudson/tasks/test/TestObject.java]: That's a bit much, isn't it?

Comment (wolfs) [inline src/main/java/hudson/tasks/test/TestObject.java]: Suggest a value, I will use tree=... anyway.

**Comment by oleg-nenashev** [comment]
It cannot be done in such way => :-1: for the merge

<code>TestAction</code> is not an <code>ExportedBean</code>. If the getter provides a list with a child, which does not has <code>ExportedBean</code> annotation, the API will return a horrible exception instead of the original response. It may lead to serious regressions in plugins depending on JUnit functionality.

There is a similar story in Jenkins core: https://github.com/jenkinsci/jenkins/pull/1698

**Comment by wolfs** [comment]
@oleg-nenashev How does https://github.com/jenkinsci/jenkins/blob/9c443c8d5bafd63fce574f6d0cf400cd8fe1f124/core/src/main/java/hudson/model/Actionable.java#L92-L93 work, then? Action also has no @ExportedBean annotation and this is in jenkins core.

I can add an @ExportedBean annotation to TestAction, too. Would this then be okay for you?

**Comment by oleg-nenashev** [comment]
> https://github.com/jenkinsci/jenkins/blob/9c443c8d5bafd63fce574f6d0cf400cd8fe1f124/core/src/main/java/hudson/model/Actionable.java#L92-L93 work, then?

I' not sure how it works... Probably it's not being exposed by default due to requests depth or uses another Stapler version. It makes sense to evaluate it in any case. I'll do it.

> I can add an @ExportedBean annotation to TestAction, too. Would this then be okay for you?

Yep, it should work well

File: src/main/java/hudson/tasks/test/TestObject.java (hunk 1 above)
Comment (oleg-nenashev) [inline src/main/java/hudson/tasks/test/TestObject.java]: I would just leave the default visibility (3?) if Jenkins shows it in the API in such case.

**Comment by wolfs** [comment]
@oleg-nenashev better now?

**Comment by oleg-nenashev** [comment]
Yes, it's OK for me now :+1: 
I think it makes sense to revise other actions and fields to export the important ones (e.g. getResult()), but it's a topic for another PR

**Pull request merged**